src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -513,6 +513,192 @@ .maxstack  1
             }
         }
 
+        [Fact, WorkItem(57994, "https://github.com/dotnet/roslyn/issues/57994")]
+        public void UserDefinedConversion_02()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe readonly struct S
+{
+    public static implicit operator delegate*<void*, void>(S _) => null;
+    void M()
+    {
+        // S -> delegate*<void*, void> -> delegate*<int*, void>
+        /*<bind>*/_ = (delegate*<int*, void>)new S()/*</bind>*/;
+    }
+}");
+
+            var verifier = CompileAndVerifyFunctionPointers(comp);
+            verifier.VerifyIL("S.M", @"
+{
+  // Code size       16 (0x10)
+  .maxstack  1
+  .locals init (S V_0)
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  initobj    ""S""
+  IL_0008:  ldloc.0
+  IL_0009:  call       ""delegate*<void*, void> S.op_Implicit(S)""
+  IL_000e:  pop
+  IL_000f:  ret
+}
+");
+
+            VerifyOperationTreeForTest<AssignmentExpressionSyntax>(comp, @"
+ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: delegate*<System.Int32*, System.Void>) (Syntax: '_ = (delega ... id>)new S()')
+  Left:
+    IDiscardOperation (Symbol: delegate*<System.Int32*, System.Void> _) (OperationKind.Discard, Type: delegate*<System.Int32*, System.Void>) (Syntax: '_')
+  Right:
+    IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: delegate*<System.Int32*, System.Void>, IsImplicit) (Syntax: '(delegate*< ... id>)new S()')
+      Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+      Operand:
+        IConversionOperation (TryCast: False, Unchecked) (OperatorMethod: delegate*<System.Void*, System.Void> S.op_Implicit(S _)) (OperationKind.Conversion, Type: delegate*<System.Void*, System.Void>, IsImplicit) (Syntax: '(delegate*< ... id>)new S()')
+          Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: True) (MethodSymbol: delegate*<System.Void*, System.Void> S.op_Implicit(S _))
+          Operand:
+            IObjectCreationOperation (Constructor: S..ctor()) (OperationKind.ObjectCreation, Type: S) (Syntax: 'new S()')
+              Arguments(0)
+              Initializer:
+                null
+");
+        }
+
+        [Fact, WorkItem(57994, "https://github.com/dotnet/roslyn/issues/57994")]
+        public void UserDefinedConversion_03()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe readonly struct S
+{
+    public static explicit operator delegate*<void*, void>(S _) => null;
+    void M()
+    {
+        // S -> delegate*<void*, void> -> delegate*<int*, void>
```

Comment:
Is this the code which required a change in CSharpTestBase in order to work?

Reply:
Yes.